bytecomp/translprim.ml

```diff
@@ -103,239 +104,253 @@ let get_used_primitives () =
-  "%nativeint_mod", Primitive (Pmodbint { size = Pnativeint; is_safe = Safe });
-  "%nativeint_and", Primitive (Pandbint Pnativeint);
-  "%nativeint_or", Primitive  (Porbint Pnativeint);
-  "%nativeint_xor", Primitive (Pxorbint Pnativeint);
-  "%nativeint_lsl", Primitive (Plslbint Pnativeint);
-  "%nativeint_lsr", Primitive (Plsrbint Pnativeint);
-  "%nativeint_asr", Primitive (Pasrbint Pnativeint);
-  "%int32_of_int", Primitive (Pbintofint Pint32);
-  "%int32_to_int", Primitive (Pintofbint Pint32);
-  "%int32_neg", Primitive (Pnegbint Pint32);
-  "%int32_add", Primitive (Paddbint Pint32);
-  "%int32_sub", Primitive (Psubbint Pint32);
-  "%int32_mul", Primitive (Pmulbint Pint32);
-  "%int32_div", Primitive (Pdivbint { size = Pint32; is_safe = Safe });
-  "%int32_mod", Primitive (Pmodbint { size = Pint32; is_safe = Safe });
-  "%int32_and", Primitive (Pandbint Pint32);
-  "%int32_or", Primitive  (Porbint Pint32);
-  "%int32_xor", Primitive (Pxorbint Pint32);
-  "%int32_lsl", Primitive (Plslbint Pint32);
-  "%int32_lsr", Primitive (Plsrbint Pint32);
-  "%int32_asr", Primitive (Pasrbint Pint32);
-  "%int64_of_int", Primitive (Pbintofint Pint64);
-  "%int64_to_int", Primitive (Pintofbint Pint64);
-  "%int64_neg", Primitive (Pnegbint Pint64);
-  "%int64_add", Primitive (Paddbint Pint64);
-  "%int64_sub", Primitive (Psubbint Pint64);
-  "%int64_mul", Primitive (Pmulbint Pint64);
-  "%int64_div", Primitive (Pdivbint { size = Pint64; is_safe = Safe });
-  "%int64_mod", Primitive (Pmodbint { size = Pint64; is_safe = Safe });
-  "%int64_and", Primitive (Pandbint Pint64);
-  "%int64_or", Primitive  (Porbint Pint64);
-  "%int64_xor", Primitive (Pxorbint Pint64);
-  "%int64_lsl", Primitive (Plslbint Pint64);
-  "%int64_lsr", Primitive (Plsrbint Pint64);
-  "%int64_asr", Primitive (Pasrbint Pint64);
-  "%nativeint_of_int32", Primitive (Pcvtbint(Pint32, Pnativeint));
-  "%nativeint_to_int32", Primitive (Pcvtbint(Pnativeint, Pint32));
-  "%int64_of_int32", Primitive (Pcvtbint(Pint32, Pint64));
-  "%int64_to_int32", Primitive (Pcvtbint(Pint64, Pint32));
-  "%int64_of_nativeint", Primitive (Pcvtbint(Pnativeint, Pint64));
-  "%int64_to_nativeint", Primitive (Pcvtbint(Pint64, Pnativeint));
-  "%caml_ba_ref_1",
+let primitives_table =
+  create_hashtable 57 [
+    "%identity", Primitive (Pidentity, 1);
+    "%bytes_to_string", Primitive (Pbytes_to_string, 1);
+    "%bytes_of_string", Primitive (Pbytes_of_string, 1);
+    "%ignore", Primitive (Pignore, 1);
+    "%revapply", Primitive (Prevapply, 2);
+    "%apply", Primitive (Pdirapply, 2);
+    "%loc_LOC", Loc Loc_LOC;
+    "%loc_FILE", Loc Loc_FILE;
+    "%loc_LINE", Loc Loc_LINE;
+    "%loc_POS", Loc Loc_POS;
+    "%loc_MODULE", Loc Loc_MODULE;
+    "%field0", Primitive ((Pfield 0), 1);
+    "%field1", Primitive ((Pfield 1), 1);
+    "%setfield0", Primitive ((Psetfield(0, Pointer, Assignment)), 2);
+    "%makeblock", Primitive ((Pmakeblock(0, Immutable, None)), 1);
```

Comment:
I'm not sure about `%makeblock`

Reply:
(But if we wanted to leave it, the arity is correct)